core/src/util/builders.cpp

```diff
@@ -117,12 +116,10 @@ void Builders::buildPolygonExtrusion(const Polygon& _polygon, float _minHeight,
         size_t lineSize = line.size();
         sumIndices += lineSize * 6;
         sumVertices += (lineSize - 1) * 4;
-
-        _ctx.numVertices = sumVertices;
     }
```

Comment:
I would suggest that instead of adjusting a pre-computed vertex count, we just don't pre-compute that number in the first place. It used to make sense to predict the vertex count when we were trying to hint buffer sizes, but now there's no reason for it. We can simply count vertices as they're added and set the context vertex count to that number after we're done.